src/server/dfly_main.cc

```diff
@@ -456,103 +456,135 @@ bool ShouldUseEpollAPI(const base::sys::KernelVersion& kver) {
   return true;
 }
 
-io::Result<string> GetCGroupPath() {
+error_code GetCGroupPath(string* memory_path, string* thread_path) {
   // Begin by reading /proc/self/cgroup
 
   auto cg = io::ReadFileToString("/proc/self/cgroup");
   CHECK(cg.has_value());
 
-  // Here, we assume that CGroup v1 is being used. This
-  // is quite likely, as CGroup v2 was introduced back in 2016.
-
-  // strip 0::<path> into just <path>, and newline.
   string cgv = std::move(cg.value());
-  size_t pos = cgv.rfind(':');
-  if (pos == string::npos)
-    return nonstd::make_unexpected(make_error_code(errc::not_supported));
 
-  string res = cgv.substr(pos + 1);
-  absl::StripTrailingAsciiWhitespace(&res);
+  // Next, depending on cgroup version we either read:
+  // N:<cgroup name>:<path> -- in case of v1, in many lines
+  // 0::<cgroup name> -- in case of v2, in a single line
 
-  return res;
+  auto stripped = absl::StripAsciiWhitespace(cgv);
+
+  vector<string_view> groups = absl::StrSplit(stripped, '\n');
+
+  if (groups.size() == 1) {
+    // for v2 we only read 0::<name>
+    size_t pos = cgv.rfind(':');
+    if (pos == string::npos)
+      return make_error_code(errc::not_supported);
+
+    string res = cgv.substr(pos + 1);
+    absl::StripTrailingAsciiWhitespace(&res);
```

Comment:
Not too critical, but if you're using `string_view`s, there's a `StripTrailingAsciiWhitespace()` that works on that, but as a bonus it will make the code (IMHO) slightly cleaner:

```cpp
string_view res = absl::StripTrailingAsciiWhitespace(cgv.substr(pos + 1));
```

Your call if you wanna do that

Reply:
Sounds good. This code was from before where `GetCGroupName` would return the name of the cgroup (which makes sense only in v2).

Good time to also rename the function